packages/video_player/video_player/example/integration_test/video_player_test.dart

```diff
@@ -162,5 +162,24 @@ void main() {
       await tester.pumpAndSettle();
       expect(_controller.value.isPlaying, true);
     }, skip: kIsWeb); // Web does not support local assets.
+
+    testWidgets('test Android video is rotated for landscapeRight',
+        (WidgetTester tester) async {
+      _controller =
+          VideoPlayerController.asset('assets/keyboard-landscape.mp4');
+
+      VideoPlayer player = VideoPlayer(_controller);
+      await tester.pumpWidget(player);
+
+      expect(find.byType(Transform), findsOneWidget);
+    }, skip: kIsWeb); // TODO(KyleFin): Skip for non-Android tests?
```

Comment:
@stuartmorgan do you know if there's a way I can run this test only for Android? Do CI tests run on Android? When I run the test on my machine with `flutter drive` ([docs](https://github.com/flutter/flutter/wiki/Plugin-Tests#integration-tests)), it runs in Chrome. (https://github.com/flutter/flutter/issues/32927 doesn't look too promising for finding device platform in integration tests.)

Reply:
I just realized too that when you said "test on the native side that the right metadata is sent, and on the Dart side that the metadata is handled with a transform" maybe I could just use a Dart unit test for this instead of an integration test with an actual video file.

Reply:
I created a sample test in packages/video_player/video_player/test/video_player_test.dart that verifies the rotation value is used in Transform (doesn't verify the event from Java code will actually set rotation field though).

Reply:
> @stuartmorgan do you know if there's a way I can run this test only for Android?

`flutter drive` takes the same `-d` option that `flutter run` does (I've added that to the wiki, thanks!)

> I just realized too that when you said "test on the native side that the right metadata is sent, and on the Dart side that the metadata is handled with a transform"

I was thinking a Dart unit test + a Java unit test. https://github.com/flutter/plugins/pull/4108 will help with the latter.

Reply:
Done. Added Dart and Java unit tests. (Reverted this integration test)